src/gui/qgsquerybuilder.cpp

```diff
@@ -73,6 +73,9 @@ QgsQueryBuilder::QgsQueryBuilder( QgsVectorLayer *layer,
   lblDataUri->setText( tr( "Set provider filter on %1" ).arg( layer->name() ) );
   txtSQL->setText( mOrigSubsetString );
 
+  mFilterLineEdit->setPlaceholderText( tr( "Search" ) );
+  connect( mFilterLineEdit, &QgsFilterLineEdit::textChanged, this, &QgsQueryBuilder::onTextChanged );
```

Comment:
mFilterLineEdit->setShowSearchIcon( true );

(probably needed for the expression query builder too!)

Reply:
Are you saying to replace the placeholder string with icon?

Reply:
No, I'm saying add it also so it gets the nice search/filter icon too

Reply:
ok, done.